narwhals/dataframe.py

```diff
@@ -593,28 +597,31 @@ def to_numpy(self) -> np.ndarray:
             >>> import polars as pl
             >>> import pyarrow as pa
             >>> import narwhals as nw
+            >>> import numpy as np
+            >>> from narwhals.typing import IntoDataFrame
+            >>>
             >>> df = {"foo": [1, 2, 3], "bar": [6.5, 7.0, 8.5], "ham": ["a", "b", "c"]}
             >>> df_pd = pd.DataFrame(df)
             >>> df_pl = pl.DataFrame(df)
             >>> df_pa = pa.table(df)
 
             We define a library agnostic function:
 
-            >>> @nw.narwhalify
-            ... def func(df):
+            >>> def agnostic_to_numpy(df_native: IntoDataFrame) -> np.array:
```

Comment:
I think the return type should be `np.ndarray` (`np.array` is just the function to create arrays)

Reply:
Thanks @MarcoGorelli! Addressed these.